R/lmer_multimember.R

```diff
@@ -404,5 +406,108 @@ lmerModMultiMember <-
+  #  message("Coercing object to class 'lmerModMultiMember'")
+  #  object <- as_lmerModMultiMember(object)
+  #  if(!inherits(object, "lmerModMultiMember")) {
+  #    warning("Failed to coerce object to class 'lmerModMultiMember'")
+  #    return(summary(object))
+  #  }
+  # }
+  ##############################################################################
+  memberships <- object@memberships
+  summ <- summary(as(object, "lmerMod"), ...)
+  summ$objClass <- class(object) # Used by lme4:::print.summary.lmerMod
+  summ$methTitle <- paste0(
+    summ$methTitle,
+    " with multiple membership random effects"
+  )
+  summ$memberships <- memberships
+  class(summ) <- c("summary.lmerModMultiMember", class(summ))
+
+  return(summ)
+}
+
+
+#' @title summary method for multimembership glmer model objects
+#' @param object glmerModMultiMember model object
+#' @param ... additional arguments to be passed on to summary.glmerMod
+#' @return summary of glmerModMultiMember object
+#' @export
+summary.glmerModMultiMember <- function(object, ...) {
+  if (!inherits(object, "glmerModMultiMember") &&
+    !inherits(object, "glmerMod")) {
+    stop(
+      "Cannot compute summary for objects of class: ",
+      paste(class(object), collapse = ", ")
+    )
+  }
+
+  memberships <- object@memberships
+  summ <- summary(as(object, "glmerMod"), ...)
+  summ$objClass <- class(object) # Used by lme4:::print.summary.glmerMod
+  summ$methTitle <- paste0(
+    summ$methTitle,
+    " with multiple membership random effects"
+  )
+  summ$memberships <- memberships
+  class(summ) <- c("summary.glmerModMultiMember", class(summ))
+
+  return(summ)
+}
+
+
+#' @title print method for multiple memberships summary for lmer
+#' @param x lmerModMultiMember object
+#' @param ... additional arguments to be passed on to print.summary.merMod
+#' @export
+print.summary.lmerModMultiMember <- function(x, ...) {
+  lme4:::print.summary.merMod(x, ...)
+  cat("\nGroup memberships per observation for multiple membership REs:\n")
+  multimember_sums <- purrr::map(x$memberships, Matrix::colSums)
+  print(cbind(
+    "Min. per obs." = purrr::map(multimember_sums, min),
+    "Mean per obs." = purrr::map(multimember_sums, mean),
+    "Max. per obs." = purrr::map(multimember_sums, max)
+  ))
+  invisible(x)
+}
+
+
+#' @title print method for multiple memberships summary for glmer
+#' @param x lmerModMultiMember object
+#' @param ... additional arguments to be passed on to print.summary.merMod
+#' @export
+print.summary.glmerModMultiMember <- function(x, ...) {
+  lme4:::print.summary.merMod(x, ...)
+  cat("\nGroup memberships per observation for multiple membership REs:\n")
+  multimember_sums <- purrr::map(x$memberships, Matrix::colSums)
+  print(cbind(
+    "Min. per obs." = purrr::map(multimember_sums, min),
+    "Mean per obs." = purrr::map(multimember_sums, mean),
+    "Max. per obs." = purrr::map(multimember_sums, max)
+  ))
+  invisible(x)
+}
```

Comment:
these are essentially the same thing, can you great an intermediate `merModMultiMember` in line with `lme4::merMod` and let that be called here? 

Also should an option disable displaying this stuff like lme4 can disable displaying the correlation of fixed effects. 

I think you avoid the dependency on `purrr` by using `lapply`.

Reply:
I considered this when I wrote it, but I wasn't sure what the inheritance structure should look like then.
`lmerMod` and `glmerMod` both inherit from `merMod`, I assumed they are meaningfully different beyond that, no? (I'll can check this to make sure.)

So should `merModMultiMember` then only inherit from `merMod`, and e.g. `lmerModMultiMember` from both `lmerMod` and `merModMultiMember`?

Reply:
Yes.